protocol/x/feetiers/keeper/keeper_test.go

```diff
@@ -131,6 +133,76 @@ func TestGetPerpetualFeePpm(t *testing.T) {
 	}
 }
 
+func TestGetPerpetualFeePpm_Referral(t *testing.T) {
+	tests := map[string]struct {
+		expectedTakerFeePpm int32
+		setup               func(ctx sdk.Context, affiliatesKeeper *affiliateskeeper.Keeper)
+	}{
+		"regular user, first tier, no referral": {
+			expectedTakerFeePpm: 10,
+			setup:               func(ctx sdk.Context, affiliatesKeeper *affiliateskeeper.Keeper) {},
+		},
+		"regular user, referral": {
+			expectedTakerFeePpm: 30,
+			setup: func(ctx sdk.Context, affiliatesKeeper *affiliateskeeper.Keeper) {
+				err := affiliatesKeeper.RegisterAffiliate(ctx, constants.AliceAccAddress.String(), constants.BobAccAddress.String())
+				require.NoError(t, err)
+			},
+		},
```

Comment:
Nit: Add two test cases that a user already at and larger than `StartingFeeTier` still gets correct fee tier